toolchain/parse/testdata/pointer/pointer_type.carbon

```diff
@@ -9,7 +9,9 @@
 // TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/parse/testdata/pointer/pointer_type.carbon
 
 fn F(p: i32*) -> i32* {
-  return p;
+  f32* + 0;
+  1 * bool* * 2;
+  return 3 * i32* * *p;
```

Comment:
It looks like you're using this to test operator precedence, but I think it was really intended for the pointer logic. Maybe we can use a separate test file under "operators" for `*` precedence testing?

Reply:
Oh, I see the fail test now. Sorry.